### PR TITLE
Fix :global() scoping

### LINKED ISSUE #1155: 🐛 BUG: `:global()` will not be properly removed if additional selectors applied
### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

When a `:global(foo)` is succeeded by any character/selector that is not a `space`, the final compilation will include the closing bracket `)`

### Steps to Reproduce

1. `npm init astro` using any template
2. replace `index.astro` with:
```astro
<head></head>

<style>
  p {
    color: red;
  }
  :global(foo):hover {
    background-color: red
  }
</style>

<p>test</p>
```
> P.S. `<head>` is requiered, see #1154 

3. using `<p>` as the reference, find the stylesheet in the devtools
4. Observe the 2nd CSS statement:
```css
foo):hover {
  background-color: red
}
```
Notice the not-removed bracket where the `:global(foo)` was before

### Link to Minimal Reproducible Example (Optional)

_No response_

### PR DESCRIPTION
**Pull request by drwpow**
## Changes

Fixes #1155. Basically a lazy RegEx assumed that the `:global()` selector always wraps the entire selector, when this may not be the case. 

## Testing

Tests added.

## Docs

I took a look at the existing `:global()` docs and they seem to suffice? I think this bug is more-or-less not working as expected rather than the documentation being missing or misleading.

### CODE REVIEW

**Review by FredKSchott — APPROVED** [review]
LGTM, but bonus points if you can come up with more weird edge cases to test for in the test file :)

File: packages/astro/test/astro-scoped-styles.test.js [hunk 1]
```diff
@@ -22,7 +22,13 @@ ScopedStyles('Scopes rules correctly', () => {
...
     '.class :global(ul li)': `.class.${className} ul li`, // allow doubly-scoped selectors
-    '.class:not(.is-active)': `.class.${className}:not(.is-active)`, // Note: the :not() selector can NOT contain multiple classes, so this is correct; if this causes issues for some people then it‘s worth a discussion
+    ':global(body:not(.is-light)).is-dark,:global(body:not(.is-dark)).is-light': `body:not(.is-light).is-dark,body:not(.is-dark).is-light`, // :global() can contain parens, and can be chained off of
+    ':global(.foo):global(.bar)': '.foo.bar', // more :global() shenanigans
+    '.class:global(.bar)': `.class.bar`, // this is technically a “useless“ :global() but it should still be extracted
+    '.class:not(.is-active):not(.is-disabled)': `.class.${className}:not(.is-active):not(.is-disabled)`, // Note: the :not() selector can NOT contain multiple classes, so this is correct; if this causes issues for some people then it‘s worth a discussion
+    ':hover.a:focus': `.${className}:hover.a:focus`, // weird but still valid (yes, it’s valid)
+    '*:hover': `.${className}:hover`,
+    ':not(.is-disabled).a': `.${className}:not(.is-disabled).a`, // also valid
```
Comment (drwpow) [inline packages/astro/test/astro-scoped-styles.test.js:31]: > LGTM, but bonus points if you can come up with more weird edge cases to test for in the test file :)

Done 🙂

Comment (drwpow) [reply packages/astro/test/astro-scoped-styles.test.js:31]: So for more context, there’s one really weird scenario where the final result is ambiguous:

```
:global(.bar).foo
```

Now, although logically this doesn’t make sense (if one class is global, but another class is scoped, they’d always be effectively scoped), semantically it’s correct and there are two possible ways to parse this:

1. **Scoped**: `.bar.foo.astro-XXXXXX`
2. **Global**: `.bar.foo` 

While #1 may be technically-correct (after all, part of the selector was unscoped), it’s actually deceptively hard to get right in every possible iteration of the syntax. So in favor of a much, much, much simpler solution, #2 was applied: if you use `:global()` on an element, we’ll just assume that you wanted that whole thing global. Even if you tacked weird stuff outside the parens.

This also seems to solve the spirit of #1155 as well—generated CSS behaving weirdly and resulting in broken styles.

So #2 is simpler and hopefully more predictable. Win–win!
